src/HttpClientPoolItem.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Psr\Http\Message\RequestInterface;
+use Http\Client\Exception;
+
+/**
+ * A HttpClientPoolItem represent a HttpClient inside a Pool.
+ *
+ * It is disabled when a request failed and can be reenable after a certain number of seconds
+ * It also keep tracks of the current number of request the client is currently sending (only usable for async method)
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class HttpClientPoolItem extends FlexibleHttpClient
```

Comment:
Oh, you mean like that, i think this is pretty much the same, however there is less inheritance in this case, so :+1: